crates/net/src/http/mod.rs

```diff
@@ -256,7 +257,25 @@ impl Request {
         let request =
             web_sys::Request::new_with_str_and_init(&url, &self.options).map_err(js_to_error)?;
 
-        let promise = gloo_utils::window().fetch_with_request(&request);
+        let global = js_sys::global();
+        let maybe_window =
+            Reflect::get(&global, &JsValue::from_str("Window")).map_err(js_to_error)?;
+        let promise = if !maybe_window.is_undefined() {
+            let window = global.dyn_into::<web_sys::Window>().unwrap();
+            window.fetch_with_request(&request)
+        } else {
+            let maybe_worker = Reflect::get(&global, &JsValue::from_str("WorkerGlobalScope"))
+                .map_err(js_to_error)?;
+            if !maybe_worker.is_undefined() {
+                let worker = global.dyn_into::<web_sys::WorkerGlobalScope>().unwrap();
+                worker.fetch_with_request(&request)
+            } else {
+                return Err(Error::GlooError(
+                    "Unsupported JavaScript global context".into(),
+                ));
+            }
+        };
```

Comment:
It's better to `panic!` here and document that. This is a non-recoverable error and returning Result for such error doesn't sound good

Reply:
done.